src/stores/store.tsx

```diff
@@ -136,17 +138,37 @@ class Store {
         await this.disconnect(false);
       }
 
-      let ethersProvider: ethers.providers.JsonRpcProvider;
-      if (this.networkName === 'private') {
+      let ethersProvider:
+        | ethers.providers.JsonRpcProvider
+        | ethers.providers.Web3Provider;
+
+      // In node-like environments that provide a wallet secret, use HDWallet
+      // provider
+      if (PRIVATE_KEY !== undefined) {
+        if (!INFURA_API_KEY) {
+          throw new Error('INFURA_API_KEY is required');
+        }
+
+        const web3Provider = new HDWalletProvider(
+          PRIVATE_KEY,
+          `https://${this.networkName}.infura.io/v3/${INFURA_API_KEY}`
+        );
```

Comment:
I don't see the requirement adding truffle code here.
In ethers you create a "Wallet" which acts as the signer.